### ISSUE TITLE
Garbage collection removes wrong layers

### ISSUE THREAD

**Issue by hlinnaka** [zenithdb/zenith#517]
After running `pgbench -s100 -i`, I noticed that there are a lot of segments with histories like this:
```
-rw-r----- 1 heikki heikki 10489914 Sep  1 17:03 rel_1663_13990_16396_0_93_00000000621B408B
-rw-r----- 1 heikki heikki 10489914 Sep  1 17:03 rel_1663_13990_16396_0_93_00000000855FA085
-rw-r----- 1 heikki heikki 10489914 Sep  1 17:04 rel_1663_13990_16396_0_93_00000000B2819678
-rw-r----- 1 heikki heikki 17337382 Sep  1 17:05 rel_1663_13990_16396_0_93_00000000B2819678_000000011F586678
-rw-r----- 1 heikki heikki 10489914 Sep  1 17:05 rel_1663_13990_16396_0_93_000000011F586678
```
So it looks like we've removed the delta layers between those image layers, but left behind the image layers. There's no reason those image layers are needed, so something's wrong with the garbage collection

I did some debugging, and I think the reason is that `layers.remove_historic` call removes wrong layer. When we have determined that we need to remove e.g. the delta layer `rel_1663_13990_16396_0_93_00000000621B408B`, it looks up the layer in the BTreeMap using the start LSN, and actually finds and removes `rel_1663_13990_16396_0_93_00000000621B408B_00000000855FA085 `, prematurely. It won't try to remove the file `rel_1663_13990_16396_0_93_00000000621B408B` again on later GC iterations, because it's already been removed from the layer map, even though it still exists on disk.

Long story short, `LayerMap::remove_historic` needs to identify the entry to remove more accurately, using more than just the 'start_lsn'.
